.ci/test.sh

```diff
@@ -190,21 +190,40 @@ if [[ $TASK == "gpu" ]]; then
     elif [[ $METHOD == "source" ]]; then
         cmake -DUSE_GPU=ON -DOpenCL_INCLUDE_DIR=$AMDAPPSDK_PATH/include/ ..
     fi
-elif [[ $TASK == "cuda" ]]; then
-    sed -i'.bak' 's/std::string device_type = "cpu";/std::string device_type = "cuda";/' $BUILD_DIRECTORY/include/LightGBM/config.h
-    grep -q 'std::string device_type = "cuda"' $BUILD_DIRECTORY/include/LightGBM/config.h || exit -1  # make sure that changes were really done
+elif [[ $TASK == "cuda" || $TASK == "cuda_exp" ]]; then
+    if [[ $TASK == "cuda" ]]; then
+        sed -i'.bak' 's/std::string device_type = "cpu";/std::string device_type = "cuda";/' $BUILD_DIRECTORY/include/LightGBM/config.h
+        grep -q 'std::string device_type = "cuda"' $BUILD_DIRECTORY/include/LightGBM/config.h || exit -1  # make sure that changes were really done
+    else
+        sed -i'.bak' 's/std::string device_type = "cpu";/std::string device_type = "cuda_exp";/' $BUILD_DIRECTORY/include/LightGBM/config.h
+        grep -q 'std::string device_type = "cuda_exp"' $BUILD_DIRECTORY/include/LightGBM/config.h || exit -1  # make sure that changes were really done
+        sed -i'.bak' 's/gpu_use_dp = false;/gpu_use_dp = true;/' $BUILD_DIRECTORY/include/LightGBM/config.h
+        grep -q 'gpu_use_dp = true' $BUILD_DIRECTORY/include/LightGBM/config.h || exit -1  # make sure that changes were really done
```

Comment:
Sorry for the delay. I'll handle these unresolved comments today.

Reply:
> I think it's better to handle this here

I'm not sure whether I understand your idea correctly. The old CUDA version only support double precision training. But the new CUDA version will support both double precision and single precision training. We still use single precision training in new CUDA version by default because it is faster and without hurting the accuracy. Users can specify it through `gpu_use_dp`. However, to ensure that results are identify to that on CPU (which uses double precision histograms), we need to switch to double precision training in CI test. That's why we need to replace the default `gpu_use_dp` setting here in `test.sh`.

Reply:
> We use single precision training in new CUDA version by default because it is faster without hurting the accuracy. However, to ensure that results are identical to that on CPU (which uses double precision histograms), we need to switch to double precision training in CI test. 

OK, I got it. Thanks for the explanation! Please add a short comment explaining why we need to change `gpu_use_dp` param here, in CI files. It wasn't obvious for me, so I thought that single precision isn't supported, just like it is in our current CUDA implementation. However, it's not true according to your comments.

> After trying, I found that it will further complicate the bash code since single quotes will treat everything inside literally.

OK, agree with you. Thanks for trying that!